hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3ARetryPolicy.java

```diff
@@ -214,7 +214,10 @@ protected Map<Class<? extends Exception>, RetryPolicy> createExceptionMap() {
 
     // policy on a 400/bad request still ambiguous.
     // Treated as an immediate failure
-    policyMap.put(AWSBadRequestException.class, fail);
+    RetryPolicy awsBadRequestExceptionRetryPolicy =
```

Comment:
should the normal retry policy -which is expected to handle network errors- be applied here, or something else

Reply:
correct me if I'm wrong but before our change, the response as `AWSBadRequestException` in fact is getting back with a HTTP 400 error code. It is different from other network failures that the `fail`/`RetryPolicies.TRY_ONCE_THEN_FAIL` has been applied for.